app/pages/settings/profile.jsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import ImageSelector from '../../components/image-selector';
+import putFile from '../../lib/put-file';
+
+const MAX_AVATAR_SIZE = 65536;
+const MAX_HEADER_SIZE = 256000;
+
+class CustomiseProfile extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      avatar: {},
+      avatarError: null,
+      header: {},
+      headerError: null
+    };
+    this.avatarGet();
+    this.headerGet();
+  }
+
+  avatarGet() {
+    this.props.user.get('avatar')
+    .then(([avatar]) => {
+      this.setState({ avatar });
+    });
+  }
+
+  headerGet() {
+    this.props.user.get('profile_header')
+    .then(([header]) => {
+      this.setState({ header });
+    });
+  }
+
+  handleMediaChange(type, file) {
+    const { user } = this.props;
+    const errorProp = `${type}Error`;
+    this.setState({ [errorProp]: null });
+    apiClient.post(user._getURL(type), { media: { content_type: file.type }})
+    .then(([resource]) => {
+      putFile(resource.src, file, { 'Content-Type': file.type });
+    })
+    .then(() => {
+      this[`${type}Get`] = null; //uncache the local request
+      this.forceUpdate();
+    })
+    .catch((error) => {
+      this.setState({ [errorProp]: error });
+    });
+  }
+
+  handleMediaClear(type) {
+    const errorProp = `${type}Error`;
+    this.setState({ [errorProp]: null });
+    this[`${type}Get`]
+    .then((resource) => {
+      !!resource && resource.delete();
+    })
+    .then(() => {
```

Comment:
Ditto for this block. Why was the original code rewriting component methods on-the-fly?

Reply:
I have no idea. Do you mean this also doesn't do anything?

Reply:
It does do something. It deletes the event handler then forces a re-render, effectively making the button useless after one successful click. But why?

Reply:
The original code generated a new event handler on each render.